zcash_client_sqlite/src/chain.rs

```diff
@@ -1,16 +1,80 @@
 //! Functions for enforcing chain validity and handling chain reorgs.
 use protobuf::Message;
 
-use rusqlite::params;
+use rusqlite::{params};
 
-use zcash_primitives::consensus::BlockHeight;
+use zcash_primitives::{
+    transaction::{components::Amount, TxId},
+    consensus::{self, BlockHeight},
+};
 
-use zcash_client_backend::{data_api::error::Error, proto::compact_formats::CompactBlock};
+use zcash_client_backend::{
+    data_api::error::Error, 
+    proto::compact_formats::CompactBlock,
+    address::RecipientAddress,
+    };
 
 use crate::{error::SqliteClientError, BlockDB};
 
 pub mod init;
 
+
+pub struct UnspentTransactionOutput {
+    pub address: RecipientAddress,
+    pub txid: TxId,
+    pub index: i32, 
+    pub script: Vec<u8>,
+    pub value: Amount,
+    pub height: BlockHeight,
+}
+
+pub fn get_confirmed_utxos_for_address<P: consensus::Parameters>(
+    params: &P,
+    cache: &BlockDB,
+    anchor_height: BlockHeight,
+    address: &str 
+) -> Result<Vec<UnspentTransactionOutput>,SqliteClientError> {
+    let mut stmt_blocks = cache.0.prepare(
+        "SELECT address, txid, idx, script, value_zat, height FROM utxos WHERE address = ? AND height <= ?",
+    )?;
+
+    let rows = stmt_blocks.query_map(
+        params![address, u32::from(anchor_height)],
+        |row| {
+            let addr: String = row.get(0)?;
+            let address = RecipientAddress::decode(params, &addr)
+            .   ok_or(format!(
+                "Could not interpret {} as a valid Zcash address.",
+                addr
+            )).unwrap();
```

Comment:
unwrap() should never be used, this error needs to be propagated out. Here's an example of something similar: https://github.com/zcash/librustzcash/blob/master/zcash_client_sqlite/src/wallet.rs#L79-L85